### PR TITLE
Feature: Add JWT

### PR DESCRIPTION
**Pull request by joaomfiumari**
## Summary

- Adds JWT token creation on user login (server-side)
- JWT interceptor on frontend so we can have the token added to every call done to the API
- JWT validation middleware on server-side

---
JWT: [JSON Web Tokens](https://jwt.io/introduction)

### CODE REVIEW

File: server_app/user/handler.js [hunk 1]
```diff
@@ -18,13 +20,17 @@ const userLogin = async(req, res) => {
...
     const user = await userService.login({username, password});
-  
+
     if (user == null) {
       return res.status(404).send("User not found");
     }
-  
+
+		user.token = jwt.sign({ id: user.userId }, SecretToken, {
+			expiresIn: 86400
```
Comment (joaomfiumari) [inline server_app/user/handler.js:31]: 1 day

File: server_app/model/handler.js [hunk 2]
```diff
@@ -152,11 +154,11 @@ const importModel = async (req, res) => {
...
 module.exports = router
-	.get("/", listAll)
-	.post("/", save)
-	.get("/:modelId", getById)
-	.put("/:modelId", edit)
-	.delete("/:modelId", remove)
-	.put("/:modelId/rename", rename)
-	.get("/:modelId/export", exportModel)
-	.post("/import", importModel);
+	.get("/",validateJWT, listAll)
```
Comment (joaomfiumari) [inline server_app/model/handler.js:159]: let me know if this validation is not needed in any of those endpoints, or if I missed any

**Comment by idmarjr** [comment]
So,... I'm not the best person to review this pull request 😓  
@miltonbsn can you take a look? 😄

**Review by miltonbsn — DISMISSED** [review]
Amazing job here @joaomfiumari! I am sorry I took so long to review it. 
This pull request improves drastically our session management, we really appreciate your effort.

**Review by miltonbsn — APPROVED** [review]

**Pull request merged**